src/pages/commits/utils.ts

```diff
@@ -89,6 +89,8 @@ const generateBuildVariantOptionsForTaskIconsFromState = (
           FAILED_STATUSES
         );
       }
+    } else {
+      statusesToShow = FAILED_STATUSES;
     }
   } else {
     statusesToShow = FAILED_STATUSES;
```

Comment:
Do you think it would be possible to change this block of `if-else`s to this?

```
  if (hasTasks) {
    statusesToShow = filterState.statuses;
  } else if (hasStatuses) {
    const onlyHasNonFailingStatuses =
      arrayIntersection(filterState.statuses, FAILED_STATUSES).length === 0;
    if (onlyHasNonFailingStatuses) {
      shouldShowTaskIcons = false;
    } else {
      statusesToShow = arrayIntersection(filterState.statuses, FAILED_STATUSES);
    }
  } else {
    statusesToShow = FAILED_STATUSES;
  }
```

This would still pass the tests and take away one of the `else` blocks, but all the scenarios can get confusing so I might have missed something 😵‍💫